caffe2/utils/math_cpu.cc

```diff
@@ -659,6 +665,252 @@ CAFFE2_SPECIALIZED_REDUCEMAX(int64_t)
 
 #undef CAFFE2_SPECIALIZED_REDUCEMAX
 
+namespace {
+
+void IncreaseDimsIndex(const int n, const int* dims, int* index) {
```

Comment:
IncreaseIndexInDims sounds better? Probably one line comment to briefly explain the purpose of this function.